cleanup.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+# This script does not need to be changed for certbots DNS challenge.
+# Please see the .dedynauth file for authentication information.
+
+(
+if [ ! -f .dedynauth ]; then
+    DEDYNAUTH=$(pwd)/.dedynauth
+    >&2 echo "File $DEDYNAUTH not found. Please place .dedynauth file in appropriate location."
+    exit 1
+fi
+
+source .dedynauth
+
+if [ -z "$DEDYN_TOKEN" ]; then
+    DEDYNAUTH=$(pwd)./dedynauth
+    >&2 echo "Variable \$DEDYN_TOKEN not found. Please set DEDYN_TOKEN=(your dedyn.io token) to your dedyn.io access token in $DEDYNAUTH, e.g."
+    >&2 echo ""
+    >&2 echo "DEDYN_TOKEN=d41d8cd98f00b204e9800998ecf8427e"
+    exit 2
+fi
+
+if [ -z "$DEDYN_NAME" ]; then
+    DEDYNAUTH=$(pwd)./dedynauth
+    >&2 echo "Variable \$DEDYN_NAME not found. Please set DEDYN_NAME=(your dedyn.io name) to your dedyn.io name in $DEDYNAUTH, e.g."
+    >&2 echo ""
+    >&2 echo "DEDYN_NAME=foobar.dedyn.io"
+    exit 3
+fi
+
+if [ -z "$CERTBOT_DOMAIN" ]; then
+    DEDYNAUTH=$(pwd)./dedynauth
+    >&2 echo "It appears that you are not running this script through certbot (\$CERTBOT_DOMAIN is unset). Please call with: certbot --manual-auth-hook=$0"
+    exit 4
+fi
+
+if [ ! "$(type -P curl)" ]; then
+    >&2 echo "Please install curl to use certbot with dedyn.io."
+    exit 5
+fi
+
+echo -e "\e[32mDeleting challenge token(s) ...\e[0m"
+
+# Figure out subdomain infix by removing zone name and trailing dot
+# foobar.dedyn.io gives "" while a.foobar.dedyn.io gives ".a"
+domain=.$CERTBOT_DOMAIN
+infix=${domain%.$DEDYN_NAME}
+
+# Remove leading wildcard from infix, if present
+# *.foobar.dedyn.io gives "" while *.a.foobar.dedyn.io gives ".a"
+infix=${infix#.\*}
+
+args=( \
+    '-sSLf' \
+    '--post301' \
+    '-o' '/dev/null' \
+    '-H' "Authorization: Token $DEDYN_TOKEN" \
+    '-H' 'Accept: application/json' \
+    '-H' 'Content-Type: application/json' \
```

Comment:
These lines are unnecessary for `DELETE` requests.

Reply:
Fair enough. A little too happy with the copy/paste on this one :-)